src/rgw/rgw_gc.cc

```diff
@@ -38,6 +39,16 @@ void RGWGC::initialize(CephContext *_cct, RGWRados *_store) {
     snprintf(buf, 32, ".%d", i);
     obj_names[i].append(buf);
   }
+
+  // Create GC queues
+  for (int i = 0; i < max_objs; i++) {
+    ldpp_dout(this, 20) << "RGWGC::initialize initing gc queue with name = " << obj_names[i] << dendl;
+    librados::ObjectWriteOperation op;
+    op.create(true);
+    uint64_t queue_size = 1048576, num_urgent_data_entries = 50;
+    cls_rgw_gc_init_queue(op, obj_names[i], queue_size, num_urgent_data_entries);
```

Comment:
@cbodley @mattbenjamin : I see eexist errors in rgw logs for each gc init queue call, something like this:
2019-07-24T10:36:45.279+0530 7f0313041940 20 garbage collection: RGWGC::initialize initing gc queue with name = gc.0
2019-07-24T10:36:45.279+0530 7f0313041940  1 --2- 192.168.0.15:0/1624217975 >> [v2:192.168.0.15:6800/32705,v1:192.168.0.15:6801/32705] conn(0x56514eabc900 0x56514e917b80 unknown :-1 s=NONE pgs=0 cs=0 l=1 rx=0 tx=0).connect
2019-07-24T10:36:45.279+0530 7f0313041940  1 -- 192.168.0.15:0/1624217975 --> [v2:192.168.0.15:6800/32705,v1:192.168.0.15:6801/32705] -- osd_op(unknown.0.0:1 4.1 4:9a6dd8c0:gc::gc.0:head [create,call rgw_gc.gc_init_queue] snapc 0=[] ondisk+write+known_if_redirected e16) v8 -- 0x56514ea83800 con 0x56514eabc900
2019-07-24T10:36:45.279+0530 7f02fd948700  1 --2- 192.168.0.15:0/1624217975 >> [v2:192.168.0.15:6800/32705,v1:192.168.0.15:6801/32705] conn(0x56514eabc900 0x56514e917b80 unknown :-1 s=BANNER_CONNECTING pgs=0 cs=0 l=1 rx=0 tx=0)._handle_peer_banner_payload supported=0 required=0
2019-07-24T10:36:45.280+0530 7f02fd948700  1 --2- 192.168.0.15:0/1624217975 >> [v2:192.168.0.15:6800/32705,v1:192.168.0.15:6801/32705] conn(0x56514eabc900 0x56514e917b80 crc :-1 s=READY pgs=15 cs=0 l=1 rx=0 tx=0).ready entity=osd.0 client_cookie=0 server_cookie=0 in_seq=0 out_seq=0
2019-07-24T10:36:45.282+0530 7f02fd948700  1 -- 192.168.0.15:0/1624217975 <== osd.0 v2:192.168.0.15:6800/32705 1 ==== osd_op_reply(1 gc.0 [create,call] v16'6 uv1 ondisk = -17 ((17) File exists)) v8 ==== 190+0+0 (crc 0 0 0) 0x56514dd24b00 con 0x56514eabc900

Then, I checked the osd logs, the cls init queue method is being invoked only once for each queue, but I can see multiple osd_op replies, the first one of which returns 0 and others return EEXISTS error:
2019-07-24T10:36:44.551+0530 7f7407631700 10 osd.0 pg_epoch: 16 pg[4.1( v 15'1 (0'0,15'1] local-lis/les=13/14 n=1 ec=13/13 lis/c 13/13 les/c/f 14/14/0 13/13/13) [0] r=0 lpr=13 crt=15'1 lcod 0'0 mlcod 0'0 active+clean]  sending reply on osd_op(client.4145.0:3 4.1 4:9a6dd8c0:gc::gc.0:head [create,call rgw_gc.gc_init_queue] snapc 0=[] ondisk+write+known_if_redirected e14) v8 0x55e3b80ba2c0
2019-07-24T10:36:44.551+0530 7f7407631700  1 -- [v2:192.168.0.15:6800/32705,v1:192.168.0.15:6801/32705] --> 192.168.0.15:0/818662822 -- osd_op_reply(3 gc.0 [create,call rgw_gc.gc_init_queue] v15'1 uv1 ondisk = 0) v8 -- 0x55e3b80ba2c0 con 0x55e3b7dd4d00

2019-07-24T10:36:44.703+0530 7f7407631700 10 osd.0 pg_epoch: 16 pg[4.1( v 16'2 (0'0,16'2] local-lis/les=13/14 n=1 ec=13/13 lis/c 13/13 les/c/f 14/14/0 13/13/13) [0] r=0 lpr=13 crt=15'1 lcod 15'1 mlcod 15'1 active+clean]  sending commit on osd_op(client.4159.0:1 4.1 4:9a6dd8c0:gc::gc.0:head [create,call rgw_gc.gc_init_queue] snapc 0=[] ondisk+write+known_if_redirected e16) v8 0x55e3b83be840
2019-07-24T10:36:44.703+0530 7f7407631700  1 -- [v2:192.168.0.15:6800/32705,v1:192.168.0.15:6801/32705] --> 192.168.0.15:0/4038767539 -- osd_op_reply(1 gc.0 [create,call rgw_gc.gc_init_queue] v16'2 uv1 ondisk = -17 ((17) File exists)) v8 -- 0x55e3b83be840 con 0x55e3b5af7600
....
2019-07-24T10:36:45.281+0530 7f7407631700 10 osd.0 pg_epoch: 16 pg[4.1( v 16'6 (0'0,16'6] local-lis/les=13/14 n=1 ec=13/13 lis/c 13/13 les/c/f 14/14/0 13/13/13) [0] r=0 lpr=13 crt=16'4 lcod 16'5 mlcod 16'5 active+clean]  sending commit on osd_op(client.4220.0:1 4.1 4:9a6dd8c0:gc::gc.0:head [create,call rgw_gc.gc_init_queue] snapc 0=[] ondisk+write+known_if_redirected e16) v8 0x55e3b8409340
2019-07-24T10:36:45.281+0530 7f7407631700  1 -- [v2:192.168.0.15:6800/32705,v1:192.168.0.15:6801/32705] --> 192.168.0.15:0/1624217975 -- osd_op_reply(1 gc.0 [create,call rgw_gc.gc_init_queue] v16'6 uv1 ondisk = -17 ((17) File exists)) v8 -- 0x55e3b8409340 con 0x55e3b7dd5a80

It looks like the last osd_op_reply is being received by rgw gc. Can you help me understand what is happening here?
my vstart cluster has 3 mons, 1 osd, 1 rgw.

Reply:
the first osd op is coming from radosgw-admin, which also calls RGWGC::initialize() through RGWRados::init_complete()

Reply:
But I see this upon bringing up the vstart cluster, I haven't executed any radosgw-admin command

Reply:
part of vstart.sh is to run `radosgw-admin user create` commands to create users for s3tests (see do_rgw_create_users)

Reply:
Ok thanks, I get it now.